### DISCUSSION TITLE
Stex file conversion

### DISCUSSION THREAD

**Discussion by TimeStop12** [ImageMagick/ImageMagick#5879] [Help]
I downloaded ImageMagick for the purpose of converting Godot Stex files back into regular Png files. When i tried to do the conversion though the .dll file to do so was not there. IM_MOD_RL_STEX_.dll of which it tried to find but could not. Looking here in the github and on the site it does not mention anything about stex. Could i get some help please

Thanks.

**Comment by snibgo** [comment]
"stex" isn't mentioned in IM documentation at https://www.imagemagick.org/script/formats.php , or in "magick -list format".

**Reply by TimeStop12** [reply]
could it have been removed? It clearly references a .dll file it expected to find at one point.

**Comment by snibgo** [comment]
What version of IM, on what platform? What was your command?

The changelog doesn't mention a "stex" format. Do you have a link to documentation?

Acording to https://fileinfo.com/extension/stex , a stex file is simply a PNG file with an added ".stex" file extension. If so, then prepending "PNG:" to the input file should allow IM to read it:
```
magick PNG:yourfile.stex out.png
```
If that doesn't help, please link to sample STEX files.

**Reply by TimeStop12** [reply]
Im using the latest release version of IM for windows. the command was the simple convert for one file format to another. Stex however is a Godot invention to take png files and turn them into streamable data. Which is apparently a pain in the ass to convert back into png. As for appending PNG: to try and convert it well this is the result. 
magick PNG:faces-c.png-8c090584e54b54ff5ac8d20269d1c270.etc2.stex faces-c.png-8c090584e54b54ff5ac8d20269d1c270.etc2.png
magick.exe: unable to open image 'faces-c.png-8c090584e54b54ff5ac8d20269d1c270.etc2.stex': No such file or directory @ error/blob.c/OpenBlob/3570.
Stex files have an extra bit of data at the start of the file too which looks to make it only readable by Godot and from what i have tested myself there is no way to export files from Godot back into png. Its a one way import. Here is the Stex files for you to play with. 
[Stex Files.zip](https://github.com/ImageMagick/ImageMagick/files/10217917/Stex.Files.zip)

**Comment by snibgo** [comment]
As you say, looking at the start of `faces-c.png-8c090584e54b54ff5ac8d20269d1c270.etc2.stex`, it is different to a PNG file. So a STEX file isn't simply a PNG with a different extension.

If there is some utility program that converts STEX to something that IM can read, that can be used by IM as an external delegate. I can show you how to do that.

If there is documentation on the STEX format, than an internal delegate ("coder") could be written for IM, but that would be more work.

**Reply by TimeStop12** [reply]
I have tried 3 different converters to try and get a usable .png file out and none of them have been able to produce a file that can be opened  properly. i'll link the one i tried first to try and convert the stex back into png format. 
https://github.com/ClementineAccount/godot-stex-to-png
It does convert it to png but it cannot be opened for some unfathonable reason. Godot is open source and does have a respository here on Github. Though im not exactly the best at interpreting what would be used to figure out how they creat the Stex format.
